### PR TITLE
Enable golint on travis, fixing all issues 

### LINKED ISSUE #209: (Easy-fix): make golint passing for each file
Remove the blacklisted file with a good fix for golint.

https://github.com/dmacvicar/terraform-provider-libvirt/pull/208/files#diff-995237d4b974eb43370299698c4a9a9bR22

### LINKED ISSUE #224: (Easy-fix):  create a constant string for error libvirt connection was nil
```console
grep -i "the libvirt connection was nil" -R 
resource_libvirt_network.go:		return false, fmt.Errorf("the libvirt connection was nil")
resource_libvirt_network.go:		return fmt.Errorf("the libvirt connection was nil")
resource_libvirt_network.go:		return fmt.Errorf("the libvirt connection was nil")
resource_libvirt_network.go:		return fmt.Errorf("the libvirt connection was nil")
resource_libvirt_network.go:		return fmt.Errorf("the libvirt connection was nil")
resource_libvirt_volume.go:		return fmt.Errorf("The libvirt connection was nil.")
resource_libvirt_volume.go:		return fmt.Errorf("The libvirt connection was nil.")
resource_libvirt_volume.go:		return fmt.Errorf("The libvirt connection was nil.")
resource_libvirt_domain.go:		return false, fmt.Errorf("the libvirt connection was nil")
resource_libvirt_domain.go:		return fmt.Errorf("the libvirt connection was nil")
resource_libvirt_domain.go:		return fmt.Errorf("the libvirt connection was nil")
resource_libvirt_domain.go:		return fmt.Errorf("the libvirt connection was nil")
resource_libvirt_domain.go:		return fmt.Errorf("the libvirt connection was nil")
resource_cloud_init.go:		return fmt.Errorf("the libvirt connection was nil")
resource_cloud_init.go:		return fmt.Errorf("the libvirt connection was nil")
resource_cloud_init.go:		return fmt.Errorf("the libvirt connection was nil")
resource_libvirt_coreos_ignition.go:		return fmt.Errorf("the libvirt connection was nil")
resource_libvirt_coreos_ignition.go:		return fmt.Errorf("the libvirt connection was nil")
```

### PR DESCRIPTION
**Pull request by MalloZup**
- enable golint (not used in travis yet)
- fix some style issues

fix #224, fix #209

### CODE REVIEW

**Comment by MalloZup** [comment]
https://github.com/dmacvicar/terraform-provider-libvirt/issues/209

**Comment by MalloZup** [comment]
@flavio  i am done with this. 

How it works:
- at moment this will not make travis failing.
- we can catch new regressions.
- we have a blacklist from files that we need to fix.

Feel free to ping me for any question/suggestion. For me this PR is done from my side

**Comment by MalloZup** [comment]
wip only for  waiting how goveralls react to modification

**Comment by MalloZup** [comment]
ready to be reviewed thx :sunflower:

**Comment by monstermunchkin** [comment]
This can be merged as soon as the conflicts have been resolved.

**Comment by monstermunchkin** [comment]
I just ran `golint` on this branch, and there are still a lot of complaints. These should be fixed if possible.

**Comment by MalloZup** [comment]
@monstermunchkin i was plaining to don't do all `golint` complaints in 1 pr, but split it in several. (that is why i disabled `golint` in travis)

i can fix everything in one pr, for me is no pb

**Comment by monstermunchkin** [comment]
Fixing it in one single PR makes more sense to me.

**Comment by MalloZup** [comment]
ok thx

**Comment by MalloZup** [comment]
thx in advance for reviewing it. now it is ready

File: libvirt/resource_libvirt_domain.go [hunk 1]
```diff
@@ -224,17 +226,17 @@ func resourceLibvirtDomainCreate(d *schema.ResourceData, meta interface{}) error
 	}
 
 	if ignition, ok := d.GetOk("coreos_ignition"); ok {
-		//var fw_cfg []defCmd
-		var fw_cfg []libvirtxml.DomainQEMUCommandlineArg
+		//var fwCfg []defCmd
```
Comment (monstermunchkin) [inline libvirt/resource_libvirt_domain.go]: This line can be removed. We don't use it.

File: libvirt/utils.go [hunk 2]
```diff
@@ -57,7 +64,7 @@ func xmlMarshallIndented(b interface{}) (string, error) {
 	return buf.String(), nil
 }
 
-// Remove the volume identified by `key` from libvirt
+// RemoveVolume Remove the volume identified by `key` from libvirt
```
Comment (monstermunchkin) [inline libvirt/utils.go]: RemoveVolume removes …

File: libvirt/utils_net.go [hunk 3]
```diff
@@ -15,6 +15,7 @@ const (
 	maxIfaceNum = 100
 )
 
+// RandomMACAddress mac random
```
Comment (monstermunchkin) [inline libvirt/utils_net.go]: RandomMACAddress returns a randomized MAC address

File: libvirt/utils_net.go [hunk 4]
```diff
@@ -36,6 +37,7 @@ func RandomMACAddress() (string, error) {
 		buf[0], buf[1], buf[2], buf[3], buf[4], buf[5]), nil
 }
 
+// RandomPort return randomport
```
Comment (monstermunchkin) [inline libvirt/utils_net.go]: RandomPort returns a random port

File: libvirt/utils_net.go [hunk 5]
```diff
@@ -44,6 +46,7 @@ func RandomPort() int {
 	return rand.Intn(maxPort-minPort) + minPort
 }
 
+// FreeNetworkInterface free network interface
```
Comment (monstermunchkin) [inline libvirt/utils_net.go]: FreeNetworkInterface returns a free network interface

File: libvirt/utils_net.go [hunk 6]
```diff
@@ -55,7 +58,7 @@ func FreeNetworkInterface(basename string) (string, error) {
 	return "", fmt.Errorf("could not obtain a free network interface")
 }
 
-// Calculates the first and last IP addresses in an IPNet
+// NetworkRange Calculates the first and last IP addresses in an IPNet
```
Comment (monstermunchkin) [inline libvirt/utils_net.go]: NetworkRange calculates …

**Comment by MalloZup** [comment]
@monstermunchkin thx for the review. I will adress you comments.

I'm opening an new issue for your remark. :+1:

File: libvirt/resource_libvirt_domain.go (hunk 1 above)
Comment (MalloZup) [reply libvirt/resource_libvirt_domain.go]: yp

File: libvirt/cloudinit_def.go [hunk 7]
```diff
@@ -19,10 +19,13 @@ import (
 	"gopkg.in/yaml.v2"
 )
 
-// names of the files expected by cloud-init
+// USERDATA is the name of file expected by cloud-init
```
[the remark this answers is not in the document]
Comment (MalloZup) [reply libvirt/cloudinit_def.go]: :rofl:

**Comment by MalloZup** [comment]
done :sunflower:

**Comment by MalloZup** [comment]
This is ready for review

File: libvirt/stream.go [hunk 8]
```diff
@@ -18,6 +20,7 @@ func (sio *StreamIO) Write(p []byte) (int, error) {
 	return sio.Stream.Send(p)
 }
 
+// Close  StreamIO
```
Comment (monstermunchkin) [inline libvirt/stream.go]: One space too many. Also, the comment could provide a bit more information, e.g. "Close closes the stream."

File: libvirt/stream.go [hunk 9]
```diff
@@ -18,6 +20,7 @@ func (sio *StreamIO) Write(p []byte) (int, error) {
 	return sio.Stream.Send(p)
 }
 
+// Close closes the stream O
```
Comment (monstermunchkin) [inline libvirt/stream.go]: stream O?

**Comment by monstermunchkin** [comment]
Travis is complaining. Please fix.

**Comment by MalloZup** [comment]
@monstermunchkin thx for the patience :sunflower:

**Pull request merged**